packages/hardhat/test/e2e/crab-Migration.ts

```diff
@@ -0,0 +1,445 @@
+            expect(d2EthBalanceAfter.sub(d2EthBalanceBefore).gte(ethToGet.sub(ethSpentToBuyBack))).to.be.true
+        })
+
+        it("d2 Claim and flash withdraw 100 percent of the tokens", async () => {
+            const constractSharesBefore = await crabStrategyV2.balanceOf(crabMigration.address);
+            const d2sharesInMigrationBefore = await crabMigration.sharesDeposited(d2.address)
+            const d2sharesV2ToMigrate = wdiv(wmul(d2sharesInMigrationBefore, totalCrabV1SharesMigrated), totalV2SharesReceived)
+            const d2EthBalanceBefore = await provider.getBalance(d2.address);
+
+            const [ethToGet, sqthToSell] = await getV2SqthAndEth(d2sharesV2ToMigrate);
+            const sqthPrice = await oracle.getTwap(wethOsqthPoolAddress, squeethAddress, wethAddress, 1, false)
+            const ethSpentToBuyBack = wdiv(wmul(wmul(sqthToSell, sqthPrice), BigNumber.from(105)), BigNumber.from(100))
+
+
+            await crabMigration.connect(d2).claimAndWithdraw(d2sharesV2ToMigrate, ethSpentToBuyBack) // Set ETH slippage higher!
+            const d2sharesInMigrationAfter = await crabMigration.sharesDeposited(d2.address)
+            const constractSharesAfter = await crabStrategyV2.balanceOf(crabMigration.address);
+            const d2EthBalanceAfter = await provider.getBalance(d2.address);
+
+            expect(d2sharesInMigrationAfter).to.be.equal("0")
+            expect(constractSharesAfter).to.be.equal(constractSharesBefore.sub(d2sharesV2ToMigrate))
+            // Check if minimum ETH is returned
+            expect(d2EthBalanceAfter.sub(d2EthBalanceBefore).gte(ethToGet.sub(ethSpentToBuyBack))).to.be.true
+        })
+
+        it("d1 should not be able to deposit after migration", async () => {
+            await expect(crabMigration.connect(d1).depositV1Shares(deposit1Amount)).to.be.revertedWith("M2");
+        })
+    })
+
+    describe("Individual claim after migration", () => {
+        let vaultIdV1: BigNumber
+        let collatV1: BigNumber
+        let collatV2: BigNumber
+        let shortV1: BigNumber
+        let shortV2: BigNumber
+        let crabV1Supply: BigNumber
+        let crabV2Supply: BigNumber
+        let crabV1SharesBefore: BigNumber
+        let crabV2ShareBefore: BigNumber
+        let userEthBalance: BigNumber
+
+        const initialize = async (address: string) => {
+            vaultIdV1 = await crabStrategyV1.vaultId();
+            [, , collatV1, shortV1] = await crabStrategyV1.getVaultDetails();
+            [, , collatV2, shortV2] = await crabStrategyV2.getVaultDetails();
+            crabV1Supply = await crabStrategyV1.totalSupply()
+            crabV2Supply = await crabStrategyV2.totalSupply()
+            crabV1SharesBefore = await crabStrategyV1.balanceOf(address)
+            crabV2ShareBefore = await crabStrategyV2.balanceOf(address)
+            userEthBalance = await provider.getBalance(address)
+        }
+
+        const increaseCR1 = async () => {
```

Comment:
this is fine for now, do we need to also test with diff CRs if there are any rounding issues?